### ISSUE TITLE
New operator: Hospice Use

### ISSUE THREAD

**Issue by jenniferduryea** [outcomesinsights/conceptql#197]
This is based on the brain-dump emailed on 9/13/18 "Re: GDM operators for generic events".

Use case:  time to hospice use from index
Definition:  
-  Will be defined using facility claims

Proposal:  one ConceptQL operator that:
- GDM compatibility
   - presence of an admission record
   - need to check associated context source_type_concept_id for "hosp"
- OMOP v4.5+ - no thoughts now
 

Tagging @markdanese for refinement (if any) and @aguynamedryan for implementation.  Scheduling for Chisel.

**Comment by markdanese** [comment]
It is UB04 based, so let's use Hospitalization (#194).  There may be a lot of NULL in this output (@aguynamedryan please make sure to test LOS calculations when discharge date is NULL).

**Comment by aguynamedryan** [comment]
What should LOS be when discharge date is NULL?

**Comment by markdanese** [comment]
I am guessing NULL.  If there is no discharge date then we can't calculate anything.  Unless we somehow know that admit and discharge are on the same day.  @jenniferduryea will have to comment but I believe we are ok with this based on our previous conversation.

**Comment by markdanese** [comment]
Just to clarify the above, hospice is billed monthly, so this makes sense.

**Comment by jenniferduryea** [comment]
I need data that contains hospice records to test this in chisel.jsaw.io.  Removing from testing and sending back to @marc-outins for implementation.

**Comment by jenniferduryea** [comment]
We need to update this operator since the GDM stores info in different spots. Please see updated specs below:

### Logic for GDM

- `collection_records.admission_details` is not null
- `contexts.source_type_concept_id` is concept mapped to "hospice" concept

### Logic for OMOPv4.5+

- Returns no records

## Output

We will use the same type of output designated for the Hospitalization operator: https://github.com/outcomesinsights/conceptql/issues/194


Sending back to @aguynamedryan for implementation.

**Comment by jenniferduryea** [comment]
Sending back to me, since Jigsaw looks like it is behaving as it should. But the ticket documentation is not updated.  Pending @marc-outins to add collections.admission_detail_id to gdm_250_extended.

**Comment by marc-outins** [comment]
@jenniferduryea gdm_250_extended_20190708 should have the fixed collections.admission_detail_ids

**Comment by jenniferduryea** [comment]
Tested this operator as an index algorithm and the operator. I'm confused.  It seems that Jigsaw is ignore admission_details records that do not have an end_date.  Comments above show that LOS and other information cannot be calculated for hospice records with a missing end_date (i.e. discharge date in the admission_details table).  But, Jigsaw should still be seeing and using information where the discharge_date is missing.  I cannot find these records when using this operator.  I think this is a bug in Conceptql.  We probably need to assign an end_date to these records for Conceptql to work.  I would suggest using admit_date = start_date = end_date, similar to how we use the patient's birthdate as start/end dates.  Sending back to @aguynamedryan  for help.  

However, I wanted to point out, that when this operator is used as an outcome or inclusion or baseline section, there may be some unintended consequences.  The admission_details table shows the admit and discharge date for every hospice claim.  So if a patient was in hospice for 3 months, there will be three admission_details records with the same admit date.  Jigsaw uses those dates to place these records in patient baseline and followup periods (assuming the bug in the first paragraph if fixed).  Here's where things get wonky.  When using hospice operator as an outcome and you have an index dated on or after the patient's admit date, no hospice records will be output as Outcomes, since all admit dates are dated on or prior to the patient's index date.  So if the patient is currently in hospice, and we find an index within their hospice admission (i.e. the three month visit), no hospice records will be output as an outcome because the start date (which is the admit date for the admission_details table) is always dated in the patient's baseline, even those the hospice discharge date is dated in the followup period.

Similarly, if the hospice operator is used as an inclusion or baseline algorithm, the hospice admit dates are all dated prior to or on the patient's index date.  However, the discharge date (i.e. end date) for these hospice records can and most likely fall into the patient's followup period.  Once the bug above is fixed, Jigsaw should recognize the admission_details records with missing end_dates. But, the hospice record with the completed admit and discharge dates will not be reported as inclusion or baseline events either, since the discharge date is dated after the patient's index date, and the event falls into the patient's followup.  Since the hospice event did not complete in the patient's baseline period, the event is not reported.

So basically, if the patient's index date is within the patient's hospice admission, there is no current way in Jigsaw to report the completed hospice admission_details record that contains the admit date in the patient's baseline period and the discharge date in the patient's followup period.

Tagging @markdanese for discussion and possible change to the operator's behavior.

**Comment by aguynamedryan** [comment]
@jenniferduryea, @markdanese, I can't move forward on this issue until we have a concise explanation of how the Hospice operator should behave.  Thanks!

**Comment by jenniferduryea** [comment]
closing this ticket in favor if #248, since there was much discussion about this and a rewrite is required.
